packages/store/test/test-encodeKey.js

```diff
@@ -40,12 +40,27 @@ const decodeKey = makeDecodeKey(decodeRemotable);
 
 const { comparator: compareFull } = makeComparatorKit(compareRemotables);
 
+const asNumber = new Float64Array(1);
+const asBits = new BigUint64Array(asNumber.buffer);
+const getNaN = (hexEncoding = '0008000000000000') => {
+  let bits = BigInt(`0x${hexEncoding}`);
+  bits |= 0x7ff0000000000000n;
+  if (!(bits & 0x0001111111111111n)) {
+    bits |= 0x0008000000000000n;
+  }
+  asBits[0] = bits;
+  return asNumber[0];
+};
+
+const NegativeNaN = getNaN('ffffffffffffffff');
+
 const goldenPairs = harden([
   [37n, 'p0000000002:37'],
   [-1n, 'n9999999999:9'],
   [1, 'fbff0000000000000'],
   [-1, 'f400fffffffffffff'],
   [NaN, 'ffff8000000000000'],
+  [NegativeNaN, 'ffff8000000000000'],
```

Comment:
Without canonicalization of `NaN` this would have ended up as `0` when decoding back.

Reply:
Cool! I just barely fixed it before you attacked ;)